File: wallet.js
const prompt = require('prompt-sync')();

class Wallet {
    constructor() {
        this.balance = 0;
    }

    deposit(amount) {
        this.balance += amount;
        console.log("Dépôt réussi");
    }

    withdraw(amount) {
        if (amount <= this.balance) {
            this.balance -= amount;
            console.log("Retrait réussi");
        } else {
            console.log("Fonds insuffisants");
        }
    }

    getBalance() {
        return this.balance;
    }
}

const monPortefeuille = new Wallet();

while (true) {
    console.log("Menu :");
    console.log("1. Déposer de l'argent");
    console.log("2. Retirer de l'argent");
    console.log("3. Consulter le solde");
    console.log("0. Quitter");

    const choix = prompt("Choisissez une option : ");

    switch (choix) {
        case '1':
            const montantDepot = parseFloat(prompt("Entrez le montant à déposer : "));
            monPortefeuille.deposit(montantDepot);
            break;
        case '2':
            const montantRetrait = parseFloat(prompt("Entrez le montant à retirer : "));
            monPortefeuille.withdraw(montantRetrait);
            break;
        case '3':
            console.log("Solde actuel :", monPortefeuille.getBalance());
            break;
        case '0':
            console.log("Au revoir !");
            process.exit(0);
        default:
            console.log("Option invalide. Veuillez choisir une option valide.");
    }
}

File: wallet.py
class Portefeuille:
    def __init__(self):
        self.solde = 0

    def deposer(self, montant):
        self.solde += montant
        print("Dépôt réussi")

    def retirer(self, montant):
        if montant <= self.solde:
            self.solde -= montant
            print("Retrait réussi")
        else:
            print("Fonds insuffisants")

    def consulter_solde(self):
        return self.solde

    def transfert(self, montant, portefeuille_cible):
        if montant <= self.solde:
            self.solde -= montant
            portefeuille_cible.deposer(montant)
            print("Transfert réussi")
        else:
            print("Fonds insuffisants pour effectuer le transfert")


# Exemple d'utilisation :
mon_portefeuille1 = Portefeuille()
mon_portefeuille2 = Portefeuille()

while True:
    print("Menu :")
    print("1. Déposer de l'argent")
    print("2. Retirer de l'argent")
    print("3. Consulter le solde")
    print("4. Transférer de l'argent")
    print("0. Quitter")

    choix = input("Choisissez une option : ")

    if choix == '1':
        montant_depot = float(input("Entrez le montant à déposer : "))
        mon_portefeuille1.deposer(montant_depot)
    elif choix == '2':
        montant_retrait = float(input("Entrez le montant à retirer : "))
        mon_portefeuille1.retirer(montant_retrait)
    elif choix == '3':
        print("Solde actuel :", mon_portefeuille1.consulter_solde())
    elif choix == '4':
        montant_transfert = float(input("Entrez le montant à transférer : "))
        mon_portefeuille1.transfert(montant_transfert, mon_portefeuille2)
    elif choix == '0':
        print("Au revoir !")
        break
    else:
        print("Option invalide. Veuillez choisir une option valide.")
